Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.CodeAnalysis;
+using System;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+
+namespace Generator
+{
+    [Generator]
+    public class SourceGenerator : ISourceGenerator
+    {
+        private string GetAssemblyName(SourceGeneratorContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.AssemblyName", out var assemblyName);
+            return assemblyName;
+        }
+
+        private string GetAssemblyVersion(SourceGeneratorContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.AssemblyVersion", out var assemblyVersion);
+            return assemblyVersion;
+        }
+
+        private string GetWinmdOutputFile(SourceGeneratorContext context)
+        {
+            string winmdDir = Path.Combine(Directory.GetCurrentDirectory(), "Generated Files");
+            Directory.CreateDirectory(winmdDir);
+            return Path.Combine(winmdDir, GetAssemblyName(context) + ".winmd");
+        }
+
+        private void GenerateWinMD(MetadataBuilder metadataBuilder, string outputFile)
+        {
+            Logger.Log("Writing " + outputFile);
+            var managedPeBuilder = new ManagedPEBuilder(
+                new PEHeaderBuilder(imageCharacteristics: Characteristics.ExecutableImage),

```

Comment:
you don't have to OR in Characteristics.Dll ?

Reply:
Updated and added a few more flags.